content/6_Advanced/String_Search.mdx

```diff
@@ -197,6 +197,27 @@ int main() {
 ```
 
 </CPPSection>
+<PySection>
+
+```py
+from typing import List
+
+
+def z_function(s: str) -> List[int]:
```

Comment:
this seems to do something different from the c++ code?

Reply:
honestly i think it's better to have the z function code separate from the code that handles input n stuff

but yeah this is true

Reply:
@SansPapyrus683, yep, it's different. I think my version is better cause it's isolated from any specific problem and I/O stuff. Makes it easier to understand and reuse. We should probably refactor the C++ version in a similar fashion. What do you think?

Reply:
i agree, though this impl is supposed to be used to solve that "finding periods" problem above

Reply:
I updated it to solve finding periods. Also extracted Z-algo in C++ implementation into a separate function. Let me know what you think.

AC C++ https://cses.fi/problemset/result/10765731/
AC Python https://cses.fi/problemset/result/10765787/